tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -483,18 +483,134 @@ export class WebGPUBackend extends KernelBackend {
     }
   }
 
+  private makeUniformsDataView(data: DataView): GPUBindingResource {
+    const dimensionsBuffer = this.acquireBuffer(
+        data.byteLength, GPUBufferUsage.COPY_DST | GPUBufferUsage.UNIFORM);
+    this.queue.writeBuffer(dimensionsBuffer, 0, data);
+
+    return {offset: 0, size: data.byteLength, buffer: dimensionsBuffer};
+  }
+
+  private numberOrArrayToDataView(
+      arrays: Array<{type: string; data: number | number[]}>,
+      length: number): DataView {
+    const BYTES_PER_ELEMENT = 4;
+    const uniformDataView =
+        new DataView(new ArrayBuffer(length * BYTES_PER_ELEMENT));
+
+    let dataViewIndex = 0;
+    arrays.forEach(array => {
+      const arrayData = array.data;
+
+      if (array.type !== 'int32' && array.type !== 'float32') {
+        throw new Error(`${array.type} not supported!`);
+      }
+
+      if (Array.isArray(arrayData)) {
+        if (array.type === 'int32') {
+          arrayData.forEach(d => {
+            uniformDataView.setInt32(
+                dataViewIndex * BYTES_PER_ELEMENT, d, true);
+            dataViewIndex++;
+          });
+        } else {
+          arrayData.forEach(d => {
+            uniformDataView.setFloat32(
+                dataViewIndex * BYTES_PER_ELEMENT, d, true);
+            dataViewIndex++;
+          });
+        }
+      } else {
+        if (array.type === 'int32') {
+          uniformDataView.setInt32(
+              dataViewIndex * BYTES_PER_ELEMENT, arrayData, true);
+          dataViewIndex++;
+        } else {
+          uniformDataView.setFloat32(
+              dataViewIndex * BYTES_PER_ELEMENT, arrayData, true);
+          dataViewIndex++;
+        }
+      }
+    });
+
+    return uniformDataView;
+  }
+
+  private computePadding(uniformsWithType:
+                             Array<{type: string; data: number[];}>): DataView {
+    let currentOffset = 0;
+    let padding = 0;
+    let dataViewIndex = 0;
+    const dimUniformsData: Array<{type: string; data: number[];}> = [];
+    uniformsWithType.forEach((d, i) => {
+      if (d.data.length === 0) {
+        d.data = [1];
+      }
+      // Complete std140 layout rules are documented here:
+      // tslint:disable-next-line:max-line-length
+      // https://www.khronos.org/registry/OpenGL/specs/gl/glspec45.core.pdf#page=159
+      let baseAlignment: number;
+      switch (d.data.length) {
+        case 0:
+          baseAlignment = 1;
+          break;
+        case 1:
+          baseAlignment = 1;
+          break;
+        case 2:
+          baseAlignment = 2;
+          break;
+        case 3:
+          baseAlignment = 4;
+          break;
+        case 4:
+          baseAlignment = 4;
+          break;
+        default:
+          util.assert(false, () => `Unsupported ${d.data.length}D shape`);
+      }
```

Comment:
Could replace this switch branch as shown below?
util.assert(false, () => `Unsupported ${d.data.length}D shape`);
const baseAlignment = d.data.length == 0 ? 1 : (d.data.length == 3 : 4 : d.data.length);

Reply:
Thanks @xhcao! I prefer the original, it's easily understood, and easy to support 5D/6D.